### ISSUE TITLE
Nile: decentralized, commission-free, local-economy focused Amazon alternative

### ISSUE THREAD

**Issue by marcocastignoli** [open-source-ideas/open-source-ideas#78]
I already wrote the description of the project as if it's already an established organization, but it is not. I'm searching for developers, lawyers and advertisers interested in the project!
### Why?
We live in a society in which huge corporations have the power to influence economies and trends. Today Amazon is the leading company for selling goods across the web, they [don’t pay taxes](https://www.businessinsider.com/amazon-not-paying-taxes-trump-bezos-2018-4?IR=T), they [underpay their employees](https://teamster.org/news/2018/02/amazon-workers-underpaid-micromanaged-and-under-surveillance), and they are [disrupting local economies](https://whatis.techtarget.com/definition/Amazon-effect). We think that the world deserves a better alternative, so we decide to found Nile.
###  What is Nile?
Nile is a decentralized ecommerce that promotes local stores, and it’s commission free: all the money you pay for a good goes directly to the seller. Nile is made to be, as far as possible, similar to Amazon; so, when you when you are searching for something, you’ll not even notice that you are using a decentralized system.
### How will you earn money?
The application is decentralized so we just need money for the developers, the lawyers and the advertisers, we don’t have to build and maintain a huge and expansive network infrastructure. So we’ll just earn money from advertisement, but it will not be specific for you, because we want your data to be yours, and we will accept to advertise only local products.
### How do you build a decentralized ecommerce?
Every city (or village) will have a Nile instance, the Nile instance is needed to group all the stores of your city so you can view them in a showcase like Amazon’s. All the Nile instances will talk to each other, so when you search for a product and it’s not near to you (it means it’s not in the instance of your city), you will find it anyway.
Stores will also contribute to the platform: every store will host its part of the system: when you buy a product, the computational cost of the operation is processed by the store. The store will have different ways to connect to the local Nile instance, if a store doesn’t handle lots of requests then it just need a browser to be part of the network. If the store needs to handle thousands of requests per seconds then a more powerful architecture is needed but we will help configuring it.
### What about the delivery?
Nile comes with a delivery service that works like Deliveroo: a network of cyclist that cover the local area. An algorithm calculates the best and cheapest route to deliver your goods.
Also the delivery service will be commission-free, when a good is delivered, the money for the delivery goes directly to the cyclists involved.
When you buy something from far away a cyclist can’t do the job, so the app will suggest the cheapest delivery method from a third party delivery service.

## Relevant Technology
I'm open to discuss about which language or protocol fit best this idea

## Who is this for
The project is very ambitious so everyone is welcome, of course experienced developers are needed to build the decentralized architecture

## Complexity and required time

Nile is an ambitious project, from a programming point of view, building a decentralized system is something difficult, there are a lot of concept involved. But a first version of Nile, without the complete decentralization implemented could be ready in just 3 months. Later, for the next 9 months, we could decentralized every piece of Nile while it is already online.
So in total we need 1 year to launch Nile.

### Complexity

- `[ ]` Beginner - This project requires no or little prior knowledge of the technolog(y|ies) specified to contribute to the project
- `[ ]` Intermediate - The user should have some prior knowledge of the technolog(y|ies) to the point where they know how to use it, but not necessarily all the nooks and crannies of the technology
- `[x]` Advanced - The project requires the user to have a good understanding of all components of the project to contribute

### Required time (ETA)

- `[ ]` Little work - A couple of days
- `[ ]` Medium work - A week or two
- `[x]` Much work - The project will take more than a couple of weeks and serious planning is required

---

Share on Twitter/other social media if you will. Something like

```
New project on Open Source Ideas Initiative: [Nile: decentralized, commission-free, local-economy focused Amazon alternative]
```

**Comment by FredrikAugust** [comment]
Wow, this sounds really cool :open_mouth:! Hope it works out :) Would love to join, but I'm starting my studies soon, so won't have time to work on this full time, though if you end up building it in a language I'm familiar with, I'll definitely come help out! :)

**Comment by paulcmal** [comment]
I feel like the basis for your project is already covered by [OpenBazaar](https://www.openbazaar.org/), a fully P2P marketplace with support for many cryptocurrencies. Maybe extending Openbazaar to support delivery via local bike coops (there are a few) or to support local currencies / cash (with optional third-party escrow) could be an option?

**Comment by marcocastignoli** [comment]
@paulcmal I already tried OpenBazaar, the problem that I see is about the "Getting started". The OpenBazaar server is written in Python or Go, this means that every shopper needs to have a machine with external software installed. I think that the average shopper doesn't have a running computer / server and doesn't want to install anything on his/her OS (if not on his/her smartphone). My idea is to have a central server for each city that's only used as a websocket server and to serve the showcase APIs (Each central server will communicate to the others, so users can see also shops in other instances), _so a shopper just need a browser to get started_.

**Comment by NorthFoxz** [comment]
The website can be hosted on IPFS and payments on smart contracts.
I can help with the project :D

**Comment by Peppuz** [comment]
Bella Marco! 
I love your ideas and share your thoughts about big companies, like Amazon that raise their capital exploiting locals and selling without even paying taxes.
I thought that this project should be full open source in order to get more people to contribute and share as well this Anti-Amazon propaganda. Decentralization on cities make sense, i agree, the vendor doesnt want to have too much crap to think about, so let them use the service only, paying a little fee monthly to make the service run and support economically the maintenance.

**Comment by AtHeartEngineer** [comment]
Typo: "when you when you are"

**Comment by marcocastignoli** [comment]
@VictorBjelkholm that's a great news! Maybe the store part of Nile could be found upon OpenBazaar, thanks for the suggestion. I'm gonna search more about this!

**Comment by paulcmal** [comment]
> every shopper needs to have a machine with external software installed

In my experience this is not a problem, especially when it comes to growing a local economy based on trust relationship with free-software activists. What is a major stopper though is key/data backup & recovery, which is usually not given enough thought leading to people losing access to stuff (and making them angry at p2p/selfhosting)

> My idea is to have a central server for each city

Well if you're not going to centralize, why not federate from the bottom up? Let people run their own (personal/community) instances and federate through ActivityPub or XMPP? That would avoid a single point of trust/failure, even if on a local level :)
(of course in order to fight spam it does make sense to build some sort of web-of-trust or a whitelist for federation like [peertube](https://joinpeertube.org/en/) does)

Thanks for taking the time to reply. I wish you lots of luck and courage to build your project and destroy these capitalist assholes who profit from our misery ! :fire:

**Comment by bigolol** [comment]
This has maybe a 0.0001% chance of succeeding, but damn I sure hope it does.

**Comment by paulmillr** [comment]
Learn capitalism. You're focusing on shitty problems. Most folks don't give a crap about decentralization or local economy as long as something's cheaper and more accessible to them, see [Prime does its job better than local government in Canada](https://www.cbc.ca/news/canada/north/iqaluit-amazon-prime-1.4193665). That local economy "need" arises from overfed western view.

Focus on real problems that would make the product a great competition to AMZN.

**Comment by marcocastignoli** [comment]
@paulmillr thanks for the suggestion, and I totally agree with you, except 

> You're focusing on shitty problems

 I think that the only way to make the local economy to rise, is to give the local store the tools that made Amazon a successful service. And that's exactly the purpose of this project.

**Comment by as1ndu** [comment]
I recommend you use http://foam.space for  proof of location when managing logistics

**Comment by nerdbeere** [comment]
Great Idea!

In Germany/Austria there's a startup (https://atalanda.com) that tries to solve at least parts of what you are describing. It's not decentralized but still an interesting project though.

**Comment by sardaukar** [comment]
Not a big fan of Amazon, would love to help.

**Comment by russellballestrini** [comment]
I would like to partner, when can we talk?

I drafted a blog post on Friday which details how I would tackle this problem, instead of building the market first I would build a free SaaS "value add" application for side gig and small business operators which would help them keep track of work-in-progress. This data would then be used to power recommendation engines, and a marketplace. The network would come naturally after that.

Reference to my post: https://russell.ballestrini.net/all-local-heros-need-a-gig-side-kick/

**Comment by marcocastignoli** [comment]
@russellballestrini @sardaukar @NorthFoxz @FredrikAugust This evening (CET) I will crete a group somewhere (suggestion about the platform? Better if it's open source, Telegram is ok?), and add you if you want, so we can speak easily.

**Comment by nadavdrewe** [comment]
I have already built a good portion of this in c# using websockets, mongodb, javaScript. The platform is token based, the payment mechanism runs on eos. I have working demos and a working server, plus various nsservicebus microservices handling message queing.

**Comment by marcocastignoli** [comment]
@nadavdrewe Wow! :open_mouth: That's a great news, can I add you to [the future group](https://github.com/open-source-ideas/open-source-ideas/issues/78#issuecomment-410739889)? Can you help explaining your projects?

**Comment by FredrikAugust** [comment]
@marcocastignoli we already have a slack group, but I agree with your "the more open source the better" philosophy, so telegram is fine with me.

**Comment by FredrikAugust** [comment]
And before someone points it out, I know telegram isn't entirely open source

**Comment by meyer9** [comment]
I have experience with blockchain projects and I'm interested in contributing. Please add me as well. I've worked on a fork of OpenBazaar, but I have some ideas of how to make it more accessible.

Basically, OpenBazaar forked the IPFS network for almost no reason. The OpenBazaar network could run just fine on the main IPFS network. AFAIK, OpenBazaar was waiting on https://github.com/ipfs/js-ipfs to finish DHT implementation for javascript to allow it to be accessed through a web browser.

I think the best way to develop a platform like the one you are proposing is to contribute/fork OpenBazaar and contribute to projects like js-ipfs to allow it to be used in a browser. Since OpenBazaar is open-source, if the company behind it is not going in a direction that we want, we can simply fork it. Personally, I don't agree with the monetization of their platform using tokens that they were proposing.

**Comment by nickdima** [comment]
@marcocastignoli I suggest using https://spectrum.chat It's a good compromise between a chat app and a message board. Using something like telegram would make it difficult to follow the discussions. A thread based system should be easier to follow, and searchable as well.

**Comment by marcocastignoli** [comment]
@meyer9 @Aeolun I don't like the idea to implement a cryptocurrency.

> If this is going to be built on proven services without any mention of block chain, tokens, ipfs or other unicorns I am interested.
Thanks! I totally agree with you. I think that for each problem there is a different solution, a platform like this one is huge, lots of different technologies will be involved, I don't want to advertise it focusing on just some of them.

**Comment by meyer9** [comment]
@marcocastignoli That sounds good to me. I do think cryptocurrencies need to play a role in the platform to be truly decentralized, however I do not think we should focus our efforts on using cryptocurrency as anything other than a P2P payment platform. We can also simply use already-existing cryptocurrencies. We'll chat about it later once the spectrum chat gets set up.

**Comment by marcocastignoli** [comment]
@russellballestrini @sardaukar @NorthFoxz @FredrikAugust @meyer9 @Aeolun We have the channel https://spectrum.chat/nile

**Comment by FredrikAugust** [comment]
Will check in later. On my way home.

**Comment by jesser360** [comment]
@marcocastignoli I would be more than happy to help with development.
I am currently working on my own e-commerce platform that integrates different APIs including package tracking, payment systems (stripe and blockchain), and my own structured back-end for managing orders.
I have also looked in depth into the development of decentralized e-commerce app built on ethereum and what that would look like (not an easy task). It'd be fun though!!
Please let me know how I can help!

**Comment by snajpa** [comment]
Hey guys!

I'm a cofounder of a nonprofit organization, which does exactly what this issue is asking for.

https://vpsfree.org

If you guys want, we can launch it all over the world - we are just about ready for that. Up until now, we've been running OpenVZ containers, but now we have our own stack based on NixOS + LXC + ZFS:

https://vpsadminos.org/

We are pro-opensource and fully transparent.

Currently, we have about 1300 members running 1700+ containers for local CZ/SK community here in Europe.

If there is enough interest, we can work together to grow this community and the way it does things ;)

You can reach me on Freenode.net IRC channel #vpsfree, nick snajpa - please, talk to me.

This is my life, I am working full time for the community, I am not the owner, though I lead things (until I become useless to the project, there's always a possibility to kick me out.).

Lately I've been thinking to codify what we do on some blockchain network - Cardano with IELE seems to be a fit choice at first and even second glance.

Let's do this!

**Comment by marcocastignoli** [comment]
@jesser360 Thanks a lot! Please join https://spectrum.chat/nile

**Comment by nadavdrewe** [comment]
Yes, add me to the group. I fully agree - don't use blockchain, use
traditional infrastructure. I have telegram and spectrum. My spectrum
username: deathmachine. Send me a link to the telegram group and I will join

**Comment by nadavdrewe** [comment]
Done. :)

**Comment by jjeffryes** [comment]
Adding in a system to support the exact use case described here should be entirely doable within OpenBazaar. These features are on the long-term road map, but if someone wanted to contribute to them before that, we'd be happy to help them do so.

Regarding support for web and mobile, we're working on both of those things right now. After we complete support for multiple coins in a single node those are our next two major release targets.

And if you don't like the Go implementation, you can always create a different one. The OB protocol is separate from the reference client and server. The OpenBazaar team's vision is that there will eventually be many specialized applications that use the protocol. An implementation focused on local products is a great example!

**Comment by djmrr** [comment]
This sounds stellar!

**Comment by deusofnull** [comment]
Plz add me to the group!  I would love to help!  Something like this would be a major challenge to run and operate, but a decentralized online shop is sorely needed to confront monopolies.

**Comment by snajpa** [comment]
Here's a live API of vpsFree:

https://api.vpsfree.cz/

As I was saying, we really need to integrate with some blockchain, so that we could do the "cloudy" hourly billing for resources.

A lot more info about us is on our knowledge base:

https://kb.vpsfree.org/

I think what we do could serve the purpose of "open community-run cloud" just fine - we just didn't have the luck to attract anyone outside CZ/SK willing to jump on board with us yet. It's not like we did a whole lot of searching for the right people ;)

So again, if anyone is interested, we are here, we've been doing this for almost 10 years now, it would be a logical next step for us, to go a bit more decentralized & global.

My hope is that we can bring this into existence much sooner, if we can find a way how to all work together.

**Comment by meyer9** [comment]
**Everyone can join here:** https://spectrum.chat/

**Comment by placer14** [comment]
Hi all. I'm a developer on github.com/OpenBazaar/openbazaar-go. We created OpenBazaar as an open and free protocol to support exactly the idea that you are attempting to create. We don't object to you starting anew, but at least take along some of the wisdom that we've developed over the past 5+ years working on this project.

tl;dr: OB1, the team working on OpenBazaar, would love to help.

Key takeaways:
- There are very significant problems that were being solved by running OB as an app on your local computer. We realize the friction it creates but as a project whose focus is "privacy" and "freedom", the choice of where to start was pretty clear. We'd be happy to share some of our thinking around this in chat, but we've covered the idea pretty well on our blog: [2014](https://openbazaar.org/blog/bitcoin-free-trade-and-terrible-humans/), [2017](https://openbazaar.org/blog/new-year-new-openbazaar-the-disruption-of-disruptive-technology/), [2018](https://openbazaar.org/blog/why-do-i-have-to-download-openbazaar/) to offer a few posts.
- IPFS is definitely the infrastructure you want to build on top of. 
- js-ipfs is pretty alpha still but they're (ProtocolLabs) moving quickly and the community is only growing.
- Don't be afraid to use centralized server where the technology is lacking. Those problems can/will be solved.
- Separate your protocol from your implementation for maximum community collaboration opportunities. OB1 - the company working on OpenBazaar protocol and proof-of-concept implementations on desktop, mobile and web - has spent the last several years focused on the UX and design and are beginning to open the protocol more formally now that we've got some of the rough edges worked out. If you can be thoughtful up front about these things without hindering your development speed, then do it.
- OpenBazaar (the protocol) will be a fully documented specification and eventually will break out our core libraries into usable implementations in Go (we only use python for testing) and maybe in JS later on as our web experience gets refined and explored.
- Our desktop implementation is fully open sourced and welcome to be forked for now and forever. I know a few projects have already started down this path and we're happy to support them as much as it aligns mutually with our projects. Our website and mobile implementation are likely to remain closed as we continue to explore OB1's business model for ongoing sustainability.
- OB1 runs an open Slack channel where a few derivative projects have been squatting to work out the details of using OB in their use case. This has worked well for getting quick support and answer to them when needed. Please feel free to use: http://openbazaar.slack.com
- We also run regular developer meetings which anyone can participate in to learn about our progress, interact with the core technical team, and share about your own work and challenges relating to OpenBazaar. Our updates are posted here afterward: https://www.youtube.com/channel/UCnw_d1lLMFhqkL5QdsuIInQ
- I'm one of the developers who is focused on improving community interactions and contributions. If you have suggestions, thoughts or feedback I'm all ears. Let me know how our team can help!

**Comment by dnk8n** [comment]
A lot of proof of concept work can be done using chatops. For an example, see [libretaxi](https://github.com/ro31337/libretaxi).

Then I would like something like libretaxi to be rebased on a federated system like [matrix](https://matrix.org/blog/home/).

The way I understand it local nodes (stores) in this case would be responsible for infrastructure costs. Private data secured on the nodes own databases, public data replicated over IPFS for example.

Essentially you now have a communication platform on which you can bolt on what you want.

I have an idea like yours, except for getting an Uber like service to exist out in rural areas. You might need to build in Uber like functionality to fulfill the deliveries. Also nodes could work together to track the progress of a long distance delivery.

Also, the otherwise empty delivery truck could be routed in order to maximise use of it's capacity. This is where the federated service would really be beneficial.

Also data analytics on historical data can be used for machine learning model training on the global system.

I wouldn't want this to become yet another ad-tech thing though. Think monetization can come in support contracts, verified nodes, etc.

Just brainstorming, haven't had time to thing all what I am saying all the way through. But it could be a start.

**Comment by erlend-sh** [comment]
In a similar vein as OpenBazaar, there’s also 
https://www.originprotocol.com

**Comment by Chris-Fourie** [comment]
Hi there,

this is something that I have been thinking about for a while. I have also been following OpenBazaar and really like the break down they have just given.

I am keen to get involved.

What makes sense to me in terms of incentivising a group of local store owners to maintain an instance of a system like this would be to have the ability to create a federated network of instances that is web accessible. 

So in a similar way that the people over at https://fediverse.party/ have made this for chat, I could imagine something similar for online stores. 

E.g. I sell Product_X and you sell Product_Y, each of us has a dedicated page for our own products and store but we also have a page for our federated store that lists both our products. This could allow sellers to form networks where their products have a symbiotic draw on customers. It also allows them to define the boundaries of their own selling network, so that if for example they want to only federate with sellers they believe are trustworthy, then this could give certain federated selling networks the opportunity to form a group identity and reputation within their local community. 

Some networks of sellers may want to tailor the look and feel of their sales platforms to suite their  customer niche, perhaps this direction may help with that as well.  

 Will join you over at https://spectrum.chat/nile =)

**Comment by varunpant** [comment]
Love to help

**Comment by FredrikAugust** [comment]
Please join the spectrum.chat @varunpant. We're currently in the middle of organizing, but we'll get together (all of us) and talk shortly

**Comment by stealthyim** [comment]
Hey @marcocastignoli if you want a mobile decentralized communication integration, we'd love to work with you at https://stealthy.im 👍

**Comment by steckerhalter** [comment]
too bad it's driven by ads again. haven't you understood that ads are most despicable thing of our time. fuck ads and fuck those that build stuff with ads.

**Comment by DrEloyd** [comment]
Is there an alternative to an ad based platform, perhaps adopt a co-op model to further drive the local first mentality?

**Comment by FredrikAugust** [comment]
@steckerhalter please refrain from using language like that (read the CoC). Try to convey your argument in a constructive manner instead of just bashing on ads.

**Comment by gagmaker** [comment]
Great idea!
However, please move away from github. It is microsoft now. They claimed multiple time they've changed, but really would you believe microsoft when they payed $7.5bln for the company evaluated just over $2bln?

**Comment by rapsacnz** [comment]
Funnily enough, there is an Amazon competitor in New Zealand called thenile.co.nz. It is books focused, like Amazon was in the early days. I use it for all my books now.

**Comment by nadavdrewe** [comment]
Guys guys guys... we need hours of chat and multiple diagrams. Bounded
contexts. Data models.. break down into layers. Break down into services.
You can't just extend and go. Don't rush into writing code without prep
work. Trust me.

On Aug 6, 2018 10:43 PM, "Caspar Harmer" <notifications@github.com> wrote:

Funnily enough, there is an Amazon competitor in New Zealand called
thenile.co.nz. It is books focused, like Amazon was in the early days. I
use it for all my books now.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/open-source-ideas/open-source-ideas/issues/78#issuecomment-410862361>,
or mute the thread

**Comment by FredrikAugust** [comment]
@nadavdrewe I don't think anyone's doing that. We are in the phase of planning, discussing et cetera. Nothing's being rushed.

**Comment by marcocastignoli** [comment]
Hello everybody, we decided to move on Slack. Please join us using [this link](https://join.slack.com/t/nileorg/shared_invite/enQtNDEyODAzODQwOTUxLTIyMTJhZmZlM2ZlMTdlNzQ1YWNmNDEyYjQ5ZWY5YjAxM2E2YTkwOTliZGIyZDliYzdmNzQ5NjY5YTNjMWU0ZTU)

**Comment by FredrikAugust** [comment]
Regarding the move to slack:
> The reason for doing so, and not using the slack of open-source-ideas, is because we assume we are going to need several channels to organize everyone working on the project, and this would simply clutter the slack of the initiative with Nile-related discussion/channels.

**Comment by marcocastignoli** [comment]
Tomorrow evening starting from 18:30 CET we will start writing a collaborative version of the whitepaper, everyone is welcome, just join us on Slack.

**Comment by jonhearty** [comment]
Loved seeing how many people commented on this -- shows how big of a problem it is and how important a solution is to the world.

We're working on a lot of this at Origin Protocol where we're using Ethereum and IPFS to decentralize marketplaces with a specific focus on sharing economies and the fractional usage of assets.

Our JavaScript library, Origin.js (launching on mainnet in the beginning of October), will allow developers to quickly spin up a decentralized marketplace, even if they don't know Solidity or how to write a smart contract. This will allow them to focus on their business and not re-building all of the core components necessary for a fully-functioning marketplace.

We also built our own DApp (on top of our platform) which acts as a meta-marketplace, aggregating listings from all of the partners that will build on Origin (currently [over 40](http://originprotocol.com/partners) have committed to doing so).

We've done quite a bit of work already (which you can check out at demo.originprotocol.com) and have a lot more that will be released over the coming months, all thanks to [our great team](http://originprotocol.com/team) that includes 3 of PayPal's first engineers (including one of the co-founders, [Yu Pan](https://en.wikipedia.org/wiki/Yu_Pan)) and 4 of YouTube's first employees (to be fair, 3 of them are the 3 PayPal guys and the fourth is our cofounder, [Matt](https://www.linkedin.com/in/matthewliu/)).

Happy either way that so many people are thinking about and working on this problem (and love that OpenBazaar chimed in), but just want to make sure everyone has an opportunity to see some of the other projects that are already working on this and have shipped a lot of code in case it makes sense to join forces rather than go at it separately! We are 100% open source (check out [our GitHub](github.com/originprotocol)) and do all of our communication out in the open on [our Discord](originprotocol.com/discord), including a [weekly public engineering call](meet.google.com/pws-cgyd-tqp). If anyone is interested, don't hesitate to reach out or say hello on the #engineering channel of our Discord!

I'll check out the newly created Slack channel, even though we moved from Slack (to Discord) because their product is not designed for parties that don't trust each other (see our founder's post on this [here](https://medium.com/originprotocol/moving-from-slack-to-discord-1dd2c939b77b)).

**Comment by marcocastignoli** [comment]
@jonhearty No one will ever move all the forces into a single project because I think every technology fit a better scenario. But what about creating a showcase for all the markets? So you can see markets coming from OpenBazaar and others coming from your project and others again from other protocols?

**Comment by jonhearty** [comment]
@marcocastignoli we're built on Ethereum, so hard for us to do this currently for projects built on other technologies, but philosophically I agree with your sentiment and this definitely aligns with what we're doing. We want to help standardize the way to represent listings on the blockchain and allow any marketplace to build on top of our protocols and developer tools. If everyone is speaking the same language, it will be easy to see a global view of all listings, regardless of which marketplace they're coming from or which client the user is accessing them from. The idea is that the data is public and therefore not owned and controlled by corporate monopolies like Amazon, Airbnb, Uber, etc.

Another huge advantage is that the user can own their identity (and therefore their reputation) and take it with them from one marketplace to another. This will also make it easier for marketplaces to acquire and onboard users. We chose to do this by building the [first working implementation](https://medium.com/originprotocol/managing-identity-with-a-ui-for-erc-725-5c7422b38c09) of ERC 725, the identity standard for Ethereum that was proposed by [Fabian Vogelsteller](https://twitter.com/feindura) (creator of ERC 20, Mist and web3.js).

**Comment by livoras** [comment]
It works. This is exactly the same thing we're doing in China and we've verified this model. This idea is 
still young and the scale of users are growing fast.

**Comment by milimetric** [comment]
Since you’re just bootstrapping this, you should really take a look at the SOLID platform that Tim Berners Lee is building at MIT: https://www.vanityfair.com/news/2018/07/the-man-who-created-the-world-wide-web-has-some-regrets/amp

It enables you to decouple customer data from the ecommerce service itself, and I think it would give your project a big competitive advantage in today’s more privacy-conscious world.

**Comment by NorthFoxz** [comment]
@livoras do you have the demo website?
I'm interested and want to help, btw I'm located at Taiwan.

**Comment by AdityaAnand1** [comment]
If you need a kick-ass android client, then count me in! (Although, I'd be willing to lend a hand in pretty much any part)

I actually had exactly this same idea just a couple days back and I was thinking by becoming decentralized, the venture also loses it's profitability so I wasn't sure how to approach this. But decentralization is definitely the way to go and as FOSS, a strong enough community could definitely push this out and reasonably compete with Amazon. This is how we should build all software, imagine a facebook and an uber and a google built this way. No middlemen. Just us.

**Comment by livoras** [comment]
@NorthFoxz We base on WeChat mini-program, leveraging on social network app helps a lot.

**Comment by piedoom** [comment]
Very neat concept.  I'd encourage readers to envision this in areas outside of your home country.  

I think the problem that should be most pressing is how to get people to use it.  Even without funding, there's ways to incentivize the use of a product.  I would find a more narrow demographic (e.g., villages and cities that have internet through phones but poor infrastructure) and market there first before attempting to solve every use case (which is near impossible!).  Some areas have great internet access whereas some places only use texts. 

If this is to be a viable alternative to big commerce sites, start small and focus on the least amount of friction possible - to both users, and hosters.

If you are targeting local govts and trying to persuade them to run your software, make sure you understand what resources they have available.  The same goes for if you're marketing at small businesses. 

Another hurdle is getting providers to list content.  Perhaps you can hook into common services.  I'm not familiar with market APIs but maybe you can provide integration with ones that businesses are already using?

To get users, you'll need a way to notify them that your service exists.  If you have any sort of marketing money, that can come in handy.  If you don't, it may be up to sellers to market for you (assuming they believe the benefits of using your service is worth it).

**Comment by hapishyguy** [comment]
Add me in the discussion group too

**Comment by lenormf** [comment]
Ignoring Openbazaar to create another Amazon controlled by you is a terrible idea

**Comment by marcocastignoli** [comment]
@lenormf we want to collaborate

**Comment by KOLANICH** [comment]
>Once a listing has been selected, a user can make a booking by sending payment to the booking smart contract along with the IPFS hash of the chosen listing and the desired interval to book. The smart contract will verify that the booking is valid and handle the transfer of tokens between the buyer and the seller, including the escrow of funds when applicable.

>The Origin user registry is a datastore of all Origin-enabled users. Origin users are identified by their Ethereum wallet addresses. In addition, the user registry also stores a mapping of all forms of identity verification that the user has successfully undertaken.

It looks like usage of the service is not private.

>To solve this challenge, both the booking smart contract and the bridge servers will use a common set of oracles and a shared algorithm to determine the exchange rate to be used.

How about security? What if someone compromised one or all of these oracles (under an oracle here I assumme a setup with the software retrieving the info from the websites of central banks and putting it into blockchain. So what do you mean under "diverse"? Software in a single machine takes info from different websites? Different vms with the same vulnrs? Different vms with redundancy based on different distros (to avoid malicious packages) and OSes in the same data center on the same physical machine? The same in different geo distributed datacenters? The same but the machines belonging to independent (how can it be achieved without compromising security?) parties? The same, but also dedicated physical machines using old hardware (the one before it had become common to insert backdoors everywhere), some contermeasures against netSpectre, protection from side-channel attacks like the ones described by Tromer et. al, modern free software like coreboot)?

**Comment by steckerhalter** [comment]
@FredrikAugust 
> please refrain from using language like that (read the CoC). Try to convey your argument in a constructive manner instead of just bashing on ads. E.g. provide alternatives, argue for why it's bad et cetera.

yeah, well. if I need to explain why ads are bad I'm probably wasting my time. but there are many alternatives. make a co-op and let people donate.

**Comment by paulcmal** [comment]
> Try to convey your argument in a constructive manner instead of just bashing on ads.

@FredrikAugust Please ask questions if you have any, but stop patronizing people about how they should talk. Especially when we're talking serious stuff: @steckerhalter's comment was not an insult or a personal attack against anyone, it was a strong political statement against advertisement and all the scammers and traitors making money out of this industry.

Who cares about a Code of Conduct when it's about doing the right thing? ;) Especially when you think of it, i'm sure one could do horrible things to a community while strictly following its code of conduct (i witnessed this happening in several free-software communities with fascist/masculinist scum infiltrating the community and destroying it from within).

But then, you may ask what's different between the advertising industry and any other industry. Indeed, is it any less ethical to buy your way into people's minds than it is to rip them off their time and energy in a factory? I don't think there's much difference (in terms of ethics). Fuck capitalism. Fuck private property. Fuck all people profiting from our misery. And death to the bourgeoisie and all the capitalist industry!

Long live free software, decentralized networks, and self-organized communities. Everything belongs to everyone. Now that i've addressed your proclaimed intellectual superiority, i'm more than ready to answer any questions you got.

PS: If you people want to get rid of slack, XMPP/Jabber is super easy to setup, standard, decentralized and provides all the features you need for collaboration : group chat (MUC), file transfer, federated git/svn, etc... And also has **great** mobile support thanks to [Conversations](https://conversations.im) on Android and [ChatSecure](https://chatsecure.org/) on iOS.

**Comment by marcocastignoli** [comment]
@placer14 Did you receive my email?

**Comment by marcocastignoli** [comment]
@paulcmal Hey Paul, this evening 18:30 CET we will try to write down together a plan on Slack. We could also discuss about the best way to communicate. I also don't like Slack, we should replace it.

**Comment by FredrikAugust** [comment]
@steckerhalter You don't need to explain why ads are bad, I think we all understand that the use of ads has negative sides. Personally I do not like, nor condone, the use of ads, but this is not my project, and if I disagree, I can form an argument as to why I think ads should not be used. What I was referring to is your first comment, where you just say "fuck ads and fuck those that build stuff with ads". This doesn't help the author improve his project, nor provide him with any alternative. 

@paulcmal I did not mean to patronize @steckerhalter, but I think his comment was strictly unnecessary (as I have explained above) and destructive, rather than constructive. As for your comment;

> it was not a personal attack against anyone

Yes it was; "fuck those who build stuff with ads". That would mean the people who are planning to build this with ads as the main source of revenue (at least to my understanding).

> it was a strong political statement

Perhaps it's just me, but I would not consider that a strong political statement. He said "fuck ads and fuck those who build stuff with ads". He spoke his distaste for ads. Your comment on the other hand, is a strong political statement, as you provide arguments for why ads are bad, and _explain why it's bad_.

Though I will ask you too, to stop using language such as "fuck _this_", and "fuck _that_", it makes it harder for people who are not as sure of their viewpoints to contribute in the conversation.

From the CoC:

> Using welcoming and inclusive language
> Being respectful of differing viewpoints and experiences

And finally;

> Now that i've addressed your proclaimed intellectual superiority

_Sigh_, really? I did not claim in any feasible way that I was "intellectually superior" to @steckerhalter. I'm simply trying to maintain a positive and welcoming community, and part of that job is regulating the comments. I thought the comment was not appropriate, hence I wrote my comment. That's it. This has *nothing* to do with me thinking I'm smarter than anyone else.

Hope that cleared out any possible misunderstandings. Let's continue working on this great project instead of discussing this.

**Comment by marcocastignoli** [comment]
@snajpa your vote count, but it's better if we speak about these kind of stuff this evening (or after this evening) so we can use a shared document to propose ideas.

**Comment by paulcmal** [comment]
> Your comment on the other hand, is a strong political statement, as you provide arguments for why ads are bad, and explain why it's bad.

I did not provide a single argument why ads are bad. Like i said, i would be pleased to expose actual arguments if some people want/need them, but i did not. My comment is not more constructed than the original was.

> Though I will ask you too, to stop using language such as "fuck this", and "fuck that"

Will do, thanks :)

> This has nothing to do with me thinking I'm smarter than anyone else.

Sorry, i thought otherwise. I felt like the original post had a valid point which you tried to silence with legalese blabbery. Hope that cleared out the misunderstanding on my part, too ;)

> If my vote would count, I'm for good ol' IRC for chat.

IRC/XMPP has probably all we need indeed :), except maybe topics for long-lived discussion. This could be handled by a forum such as discourse?

**Comment by FredrikAugust** [comment]
> But then, you may ask what's different between the advertising industry and any other industry. Indeed, is it any less ethical to buy your way into people's minds than it is to rip them off their time and energy in a factory? I don't think there's much difference (in terms of ethics).

I took this as an argument for why ads are bad. Correct, you're not explicitly saying what the downsides are, but I'd say it definitely outlined some of the social and ethical ramifications of advertisements and capitalism (though they are in several ways yin and yang) ("buying your way into people's minds").

Regardless, I'm glad we resolved this dispute so we can continue building cool open source projects! :muscle:

**Comment by daveloyall** [comment]
I won't be using slack.  Start an IRC channel on freenode or OFTC or some established network.  You can put a slack-bridge bot in the channel if you like.

**Comment by Thrilleratplay** [comment]
I do not mean to be a jerk, I just want to point out that, from an outsider, a majority of these comments are back and forths on how to communicate.

Do not waste too much time and enthusiasm on stuff like this.  Pick a place that will allows data migration and get started on why you are here in the first place.  Move the concersation if it is warranted later.

If no such platform exists that allows this sort of organization for open source projects, then that sounds like a good [open source idea](https://github.com/open-source-ideas/open-source-ideas).

**Comment by teresaejunior** [comment]
Bitboost already proposes a decentralized market on the ETH blockchain (with fees reduced, they have a IPFS solution regarding the Gas): http://bitboost.net

**Comment by marcocastignoli** [comment]
@daveloyall @paulcmal @snajpa we created a document so we can share our skills and set a preference about the communication tool. Please fill it if your are interested in the project https://goo.gl/forms/C2xTeJLtwUO1QL4p2

**Comment by snajpa** [comment]
@marcocastignoli well I’ve voiced my opinion, yet you’re continuing the debate in things which are least constructive.
As you appear to be the leader of this idea, you don’t need to give the people an option to vote on every single decision you’ll make. Either you have it figured out right, where the project should be and people will follow you there because of the sincerety of the plan, or you don’t lead and then we’re waiting for a chaos to self-organize into something way less entropic. Which, without a pretty specific idea of “what this is and what this should be”, will always be overrun with people with more motivation and more concrete goals.
Back in the day when we were working on vpsFree from a total zero with nothing much existing to copy as an example, we didn’t have the stones to be thinking about a open cloud platform, yet be choosing all the closed tools to even figure out, how to get anywhere from that zero.
Closed attitude would get us exactly nowhere. We started as a blog on a community portal, which by the way has had a deep respect for authorship and the rights you tend to hand out with your own texts on proprietary platforms.

I don’t mean to sound negativistic, but let me know, when you’re truly open and have the basics figured out.

Asking me to fill out another form with my behavior being tracked in the Google datacenters is certainly not a good way forward. I’ve written the same here on Github and on that Spectrum platform; sorry, but I won’t repeat it for the third time, when it wasn’t heard the first or the second time.

I don’t want to sound insulted or upset, because that wouldn’t be accurate - I just have better things to do at this moment.

For example figuring out a ZFS management utils to be workibg in user namespaces in containers, so that we can run Docker builds at a more native time without the VFS driver ;)

**Comment by marcocastignoli** [comment]
@snajpa can we please speak and you decide how to? I'm really interested to hear you suggestions, sorry if I didn't read too much messages here on github.

**Comment by FredrikAugust** [comment]
@snajpa Hello, I read through your comment, and I'm sorry this is the way you experienced things. It is true the beginning of this project was far from optimal, with hopping from GitHub to spectrum to slack, but I'd say the reason for that is relatively straight-forward. First we discussed on GitHub because that's where the idea was posted, but soon realized that it got too cluttered, and started looking for an alternative. I believe we first suggested slack, but people wanted something open-source, so we used spectrum.chat. After short time with that platform, we realized it was not appropriate for our use -- it felt too slow, and the chat feature was severely lacking. In order not to slow down the process, we moved to the original plan -- slack, because it had everything we wanted, and doesn't require any setup etc., so we could use it right away.

Sadly, this means that those rigid on not using proprietary software will not be able to join, and this is something we will address by moving to rocket.chat (or something similar) as soon as we can.

Until then, if you want to communicate, feel free to contact us on email, and we'll keep you up to speed :)

**Comment by snajpa** [comment]
Or, as I said, you can already find the whole team of vpsFree.cz on IRC, freenode.net #vpsfree ;)

Really, feel free to stop by and ask us, how we're doing things ;)

(guys, no need to apologize to me, I'm just stating as I see it from 10km foot view above => as I said, we're here, we've been doing virtual servers at nonzero scale for 10 years now, so you can just deploy what we do as a platform -> if you decide to go this way, things will move rather quickly, because I think we at vpsFree all know, what to do, by now)

**Comment by FredrikAugust** [comment]
Wonderful @snajpa! We'll definitely stop by :) Thanks for your reply!

**Comment by codekk** [comment]
sounds massively cool. Im in!

**Comment by FredrikAugust** [comment]
@codekk just join the slack channel, and skim through the chats in `#general`. That should get you up to date

**Comment by eduard-tkv** [comment]
The slack invite link is dead. I'm not an experienced developer but would still like to help.

**Comment by marcocastignoli** [comment]
@eduard-tkv updated https://join.slack.com/t/nileorg/shared_invite/enQtNDE4OTM4NDU4NzM5LWY2ZDU3Njg1ZDk0MjE5NmI2ZmM4ZTYwZjdjMzQxMzBjNTM5NTVmMzM5NzM1NDI4NWI0YTk0YjdhOTRhNjIzOGE

**Comment by marcocastignoli** [comment]
@snajpa Hey :) I know it was a long time ago but now I created the bridge you were suggesting between Slack and IRC. You can join us here irc://freenode/##nile

**Comment by madbob** [comment]
FYI: in Italy we have "[ethical purchasing groups](https://en.wikipedia.org/wiki/Gruppi_di_Acquisto_Solidale)", groups of people living on the same area and collectively buying goods mostly from local producers. Usually those are organized in associations, and periodically collect orders from members (on a weekly, monthly, biannual basis, depending on the type of product) to be then shipped together.
Smaller groups communicate via email with spreadsheet files in attachment. Larger groups use more or less specific software applications.

I maintain one of most popular applications, since 2009: [GASdotto](https://www.gasdotto.net/) is of course [opensource](https://github.com/madbob/GASdottoNG), it is [partially translated](https://hosted.weblate.org/projects/gasdottong/translations/) also in English, German and French, a free (donations supported) hosting service is provided to lesser tech savvy users to just click and run their own instance, and actually there are more than 100 freely hosted groups for a turnover of about 2 million euros per year on local economies.

Not visionary as the Nile project, but it actually exists ;-)
